sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/Conditions/ComparisonCondition.cs

```diff
@@ -35,43 +36,23 @@ public ComparisonCondition(string field, QueryComparisonOperator comparisonOpera
 
         public override string Stringify()
         {
-            // TODO -- generic type on this class to handle different inputs
             var conditionString = new StringBuilder();
             conditionString.Append($"{Field} {QueryConstants.ComparisonOperators[Operator]} ");
 
             // check to see if the input is numeric value -- if not, we need single quotes around Value
-            bool isNumeric;
-            bool quoteSurrounded = (Value[0] == '\'' && Value[Value.Length - 1] == '\'');
-
-            // check to see if the user has entered single quotes around the Value
-            if (quoteSurrounded)
+            if (typeof(T) == typeof(string))
             {
-                isNumeric = int.TryParse(Value.Substring(1, Value.Length - 2), out _);
+                // if string, surround value with single quotes to denote string type in SQL
+                conditionString.Append($"'{Value}'");
             }
-            else
-            {
-                isNumeric = int.TryParse(Value, out _);
-            }
-
-            if (!isNumeric)
+            else if (typeof(T) == typeof(int))
             {
-                if (!quoteSurrounded)
-                {
-                    // surround with single quotes
-                    Value = "'" + Value + "'";
-                }
-
+                // if integer, insert raw value into query string (nothing needs to be done)
                 conditionString.Append(Value);
             }
-            // if not numeric but surrounded by quotes, it must be a number surrounded by quotes (which must be removed)
-            else if (quoteSurrounded)
-            {
-                conditionString.Append(Value.Substring(1, Value.Length - 2));
-            }
-            // if numeric and not surround by quotes, we can add as is
             else
             {
-                conditionString.Append(Value);
+                throw new InvalidOperationException("Invalid condition: use a string or an integer.");
```

Comment:
I think you can get around the fact that you only support two types (should you only support two types? can you think of anything else that might be used in a comparison operation) by restricting the type T to Strings and Integers.